src/mots/config.py

```diff
@@ -0,0 +1,56 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Configuration classes used to initialize and manage mots in a repo."""
+import logging
+
+from datetime import datetime
+from pathlib import Path
+from ruamel.yaml import YAML
+
+logger = logging.getLogger(__name__)
+
+yaml = YAML()
+yaml.indent(mapping=2, sequence=4, offset=2)
+
+DEFAULT_CONFIG_FILEPATH = Path("./mots.yaml")
+
+
+class FileConfig:
+    """Loader and writer for filesystem based configuration."""
+
+    def __init__(self, path: Path = DEFAULT_CONFIG_FILEPATH):
+        """Initialize the configuration with provided config file path."""
+        if not path.exists() and path.is_file():
+            raise ValueError(f"{path} does not exist or is not a file.")
+        self.path = path
+        self.repo_path = path.parent
+        self.config = None
+
+    def init(self):
+        """Initialize a repo with a config file, if it does not contain it."""
+        if not self.path.is_file():
+            # File does not exist, create it.
+            now = datetime.now().isoformat()
+            self.config = {
+                "repo": str(Path(self.path).resolve().parts[-2]),
```

Comment:
Is `-2` specific to `./mots.yaml` here? Could a different path be passed that breaks this, ie via the `--path` CLI argument? Maybe it's fine since the `path` arg to `init` is for the repo itself, but just want to put it on your radar.

Reply:
So the answer is yes, it is specific to this path. Since the path to the config file is assumed to be in the top level directory of the repo, this should never cause any problems. It might be good to add some more explicit checks to make sure that `mots.yaml` is always at the top level, but basically if this is not the case, then other things will break. Agreed that this needs to evolve a little more though -- maybe once I add more support for other source control awareness (e.g. `hg files`).

Reply:
Makes sense. I won't block the review on this, but adding a TODO comment here would be nice.

Reply:
Filed [a bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1753714) for this.